src/main/kotlin/com/hibob/ex2-loops.kt

```diff
@@ -0,0 +1,37 @@
+package com.hibob
+
+fun isContainDolarSymbol(s: String): Boolean = "$" !in s
+
+fun isContainOnlyNumbers(s: String): Boolean {
+    var numberOnly = true
+    for (char in s) {
+        if (char in '0'..'9') continue else numberOnly = false
+    }
+    return !numberOnly
+}
+
+fun isValidIdentifier(s: String): Boolean {
+    // vacant
+    if (s == "" ) {
+        return false
+    }
+
+    // is first char start with number
+    if (s[0] in '0'..'9') {
+        return false
+    }
+
+    val res = isContainOnlyNumbers(s)
+    if (!res) return false
+
+    return isContainDolarSymbol(s)
```

Comment:
Why are you only checking for the dollar sign? All special characters are not allowed, except for underscore.

Reply:
I understood that only $ isn't allowed, it wasn't written explicetly, I get that I have to change. I hope to get to that ASAP.